api/rosetta/data_balance_integration_test.go

```diff
@@ -177,98 +201,542 @@ func TestGetBalance(t *testing.T) {
 			err = api.Balance(ctx)
-			test.wantHandlerErr(t, err)
+			assert.NoError(t, err)
+
+			assert.Equal(t, http.StatusOK, rec.Result().StatusCode)
 
-			// validate response data
+			// unpack response
+			var balanceResponse rosetta.BalanceResponse
+			require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &balanceResponse))
 
-			// validate status code
-			if test.wantStatusCode != http.StatusOK {
+			test.validateBlock(balanceResponse.BlockID)
 
-				// ugly workaround for HTTP status code set by router vs set on context
-				e, ok := err.(*echo.HTTPError)
-				require.True(t, ok)
-				assert.Equal(t, test.wantStatusCode, e.Code)
+			// verify that we have the same amount of balances as requested
+			if assert.Equal(t, len(balanceResponse.Balances), len(test.wantBalances)) {
 
-				// nothing more to do, response validation should only be done for '200 OK' responses
-				return
+				// verify returned balances - both the value and that the output matches the input spec
+				for i, wanted := range test.wantBalances {
+
+					balance := balanceResponse.Balances[i]
+
+					assert.Equal(t, test.request.Currencies[i].Symbol, balance.Currency.Symbol)
+					assert.Equal(t, test.request.Currencies[i].Decimals, balance.Currency.Decimals)
+					assert.Equal(t, wanted, balance.Value)
+				}
 			}
+		})
+	}
+}
 
-			assert.Equal(t, test.wantStatusCode, rec.Result().StatusCode)
+func TestBalanceErrors(t *testing.T) {
 
-			// unpack response
-			var balanceResponse rosetta.BalanceResponse
-			require.NoError(t, json.Unmarshal(rec.Body.Bytes(), &balanceResponse))
+	db := setupDB(t)
+	api := setupAPI(t, db)
 
-			// verify that the block data matches the input data
-			assert.Equal(t, test.request.BlockID.Index, balanceResponse.BlockID.Index)
-			assert.Equal(t, test.request.BlockID.Hash, balanceResponse.BlockID.Hash)
+	// defined valid balance request fields
+	var (
+		testAccount = identifier.Account{Address: "754aed9de6197641"}
 
-			// verify that we have at least one balance in the response
-			if assert.Len(t, balanceResponse.Balances, 1) {
+		testBlock = identifier.Block{
+			Index: 13,
+			Hash:  "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb23",
+		}
 
-				// verify the balance data - both the value and that the output matches the input spec
-				balance := balanceResponse.Balances[0]
-				assert.Equal(t, test.request.Currencies[0].Symbol, balance.Currency.Symbol)
-				assert.Equal(t, test.request.Currencies[0].Decimals, balance.Currency.Decimals)
-				assert.Equal(t, test.wantBalance, balance.Value)
-			}
+		validAddressSize = 2 * flow.AddressLength
+	)
+
+	const (
+		invalidAddress = "0000000000000000" // valid 16-digit hex value but not a valid account ID
+
+		trimmedBlockHash = "af528bb047d6cd1400a326bb127d689607a096f5ccd81d8903dfebbac26afb2" // block hash a character short
+
+		trimmedAddress    = "754aed9de619764"  // account ID a character short
+		invalidAddressHex = "zzzzzzzzzzzzzzzz" // invalid hex string
```

Comment:
Again, best to use a valid address with a single wrong character (maybe at the end or beginning to make it more visible).